File: client/src/main/java/au/com/codeka/warworlds/client/util/eventbus/EventHandler.java
package au.com.codeka.warworlds.client.util.eventbus;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import au.com.codeka.warworlds.client.concurrency.Threads;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface EventHandler {
  /** Which thread should the callback be called on, {@link Threads#UI}, etc? */
  Threads thread() default Threads.UI;
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/util/eventbus/EventHandler.kt
package au.com.codeka.warworlds.client.util.eventbus

import au.com.codeka.warworlds.client.concurrency.Threads

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class EventHandler(
    /** Which thread should the callback be called on, [Threads.UI], etc?  */
    val thread: Threads = Threads.UI)
